src/services/gasPrice.js

```diff
@@ -31,12 +31,20 @@ const foreignBridge = new web3Foreign.eth.Contract(ForeignABI, FOREIGN_BRIDGE_AD
 
 let cachedGasPrice = null
 
+function gasPriceWithinLimits(gasPrice) {
+  return GAS_PRICE_BOUNDARIES.MIN <= gasPrice && gasPrice <= GAS_PRICE_BOUNDARIES.MAX
+}
+
 async function fetchGasPriceFromOracle(oracleUrl, speedType) {
   const response = await fetch(oracleUrl)
   const json = await response.json()
   const gasPrice = json[speedType]
   if (!gasPrice) {
     throw new Error(`Response from Oracle didn't include gas price for ${speedType} type.`)
+  } else if (!gasPriceWithinLimits(gasPrice)) {
+    throw new Error(
+      `Response from Oracle included a gas price out of boundaries ${gasPrice} for ${speedType} type.`
```

Comment:
And, please correct me if I am wrong, we will use cached version of the gas price? Why do you think it is better than use the limit instead? For example, if the gas price oracle returns 0.5 `gwei`, we will use 1 `gwei`, if the oracle returns 300 `gwei` we will use 250 `gwei`. The reason is that if we have a real oracle's response (rather than compromised), these values will not impact much the transactions. Especially in case if the gas price goes up suddenly - the usage of the limit value will still allow to transactions included to the block whereas the fallback value will slow the transactions down.

Reply:
I think you are right, we should use the limit

Reply:
Done  https://github.com/poanetwork/token-bridge/pull/133/commits/7654da6df98826ec28f0dbca74e8c13e220cc9b4

Reply:
thanks!